sql/driver/wire.proto

```diff
@@ -40,6 +49,10 @@ message Datum {
     double float_val = 3;
     bytes bytes_val = 4;
     string string_val = 5;
+    Timestamp time_val = 6;
+    // TODO(vivek): Add additional types like Date and Interval that are
```

Comment:
So this is possible using `ValueConverter`, but it's going to suck, putting it mildly. We'd have to use our `ValueConverter` to convert e.g. `time.Duration` into some special value (which is one of the supported `driver.Value`s) so that we can identify it in `({conn|stmt}).Query` before converting to `driver.Datum`.